Shopify/cli
# Override plugin name when loading commands in CLI

### WHY are these changes introduced?
With the introduction of bundling we no longer have "plugins" internally, all commands are flattened into the @shopify/cli core plugin. This affected the `plugin` field in our analytics where everyhting is `cli` now.

This PR fixes that by overriding the plugin name for every command coming from a plugin.



### WHAT is this pull request doing?
- Set a `customPluginName` to every command coming from a plugin.
- Use tha `customPluginName` (if available) when building the analytics payload.


### How to test your changes?
- Set a breakpoint in `packages/cli-kit/src/private/node/analytics.ts`, where we read the `customPluginName`.
- Run a command like `app info`
- The `customPluginName` should contain `@shopify/apo`


### Post-release steps



### Measuring impact

How do we know this change was effective? Please choose one:


### Checklist
